docs/specs/user-model.md

```diff
@@ -312,3 +338,207 @@ Once used, a recovery code is invalidated.
+
+## Claims Mapping
+
+The claims mapping is a list in the app configuration.
+The mapping affects the claims of all users of the app.
+The mapping in the app configuration can override [builtin claims mapping](#builtin-claims-mapping) with the same `name_pointer`.
+
+### Builtin claims mapping
+
+Builtin claims mapping is a list of mapping entries with `kind: "system"` and supported standard claims name.
+
+The full list is:
+
+```
+- kind: "system"
+  name_pointer: "#/email"
+- kind: "system"
+  name_pointer: "#/email_verified"
+- kind: "system"
+  name_pointer: "#/phone_number"
+- kind: "system"
+  name_pointer: "#/phone_number_verified"
+- kind: "system"
+  name_pointer: "#/preferred_username"
+```
+
+That is, the claims `email`, `email_verified`, `phone_number`, `phone_number_verified` and `preferred_username` are supported natively without any configuration.
+
+#### Builtin claims mapping - email
+
+```
+- kind: "system"
+  name_pointer: "#/email"
+```
+
+When this mapping is present, the `email` claim of the user is the `email` standard attribute.
+
+#### Builtin claims mapping - email\_verified
+
+```
+- kind: "system"
+  name_pointer: "#/email_verified"
+```
+
+When this mapping is present, the `email_verified` claim of the user is computed using the information of [user verification](./verification.md)
+
+#### Builtin claims mapping - phone\_number
+
+```
+- kind: "system"
+  name_pointer: "#/phone_number"
+```
+
+When this mapping is present, the `phone_number` claim of the user is the `phone_number` standard attribute.
+
+#### Builtin claims mapping - phone\_number\_verified
+
+```
+- kind: "system"
+  name_pointer: "#/phone_number_verified"
+```
+
+When this mapping is present, the `phone_number_verified` claim of the user is computed using the information of [user verification](./verification.md)
+
+#### Builtin claims mapping - preferred\_username
+
+```
+- kind: "system"
+  name_pointer: "#/preferred_username"
+```
+
+When this mapping is present, the `preferred_username` claim of the user is the `preferred_username` standard attribute.
+
+### Custom attributes claims mapping
+
+```
+- kind: "custom_attributes"
+  name_pointer: "#/zoneinfo"
+  value_pointer: "#/profile/preferred_timezone"
+```
+
+Custom attributes claims mapping project the value pointed by `value_pointer` in the custom attributes into the location pointed by `name_pointer` in the claims.
+
+### Claims Mapping Example
+
+The following example illustrates the full capability of claims mapping.
+
+Given the following effective claims mapping
+
+```yaml
+mapping:
+- kind: "system"
+  name_pointer: "#/email"
+- kind: "system"
+  name_pointer: "#/email_verified"
+- kind: "system"
+  name_pointer: "#/phone_number"
+- kind: "system"
+  name_pointer: "#/phone_number_verified"
+- kind: "system"
+  name_pointer: "#/preferred_username"
+
+- kind: "custom_attributes"
+  name_pointer: "#/zoneinfo"
+  value_pointer: "#/profile/preferred_timezone"
```

Comment:
I suppose it's forbidden to set mapping for an intermediate path of any value pointer?
i.e. If there is a mapping mapped to `#/profile/preferred_timezone`, developer cannot set mapping mapped to `#/profile` or `#`.